torch/utils/_import_utils.py

```diff
@@ -29,7 +29,7 @@ def dill_available():
 
 
 @functools.lru_cache
-def import_dill():
+def import_dill() -> Optional[object]:
```

Comment:
When I try this locally I get errors in torch/_dynamo/replay_record.py (where this is used).

I'm not sure the right way to do this... maybe something like this?

```
if TYPE_CHECKING:
    import dill

@functools.lru_cache
def import_dill() -> Optional["dill"]:
```

or we can just return `Optional[Any]` which is... ick.